core/ir/instr_shared.c

```diff
@@ -3013,7 +3013,7 @@ instr_get_operation_size(instr_t *instr)
 void
 instr_set_operation_size(instr_t *instr, opnd_size_t operation_size)
 {
-    CLIENT_ASSERT(instr_get_isa_mode(instr) != DR_ISA_REGDEPS,
+    CLIENT_ASSERT(instr_get_isa_mode(instr) == DR_ISA_REGDEPS,
```

Comment:
Ah, good catch, thank you for fixing this.

Reply:
The IR code itself sets the field directly so there was no use of this until the new test.